src/clr-angular/utils/focus/key-focus/key-focus.ts

```diff
@@ -36,19 +45,50 @@ export class ClrKeyFocus {
     }
   }
 
+  get nativeElement(): HTMLElement {
+    return this.elementRef.nativeElement;
+  }
+
   get focusableItems() {
     if (this._focusableItems) {
       return this._focusableItems;
     } else {
-      return this.clrKeyFocusItems.toArray();
+      return this.clrKeyFocusItems && this.clrKeyFocusItems.toArray();
```

Comment:
nit: Why do we need this? Empty QueryList is still a QueryList, just as an empty array is still an array. Do we use it somewhere before view has finished initializing?

Reply:
This was to prevent `moveTo` method from being called if `this.clrKeyFocusItems` is empty.  If the method is called while the `this.clrKeyFocusItems` is null, it would throw an error.

Reply:
Yes, but if it's empty it's not null. It's only null before ContentChildren has initialized. Or I'm missing something?
Just curious.

Reply:
Yes, but it's a guard against the case when `focusableItems` gets accessed even before the `KeyFocus` directive is able to register its QueryList because `QueryList` is `undefined` until `AfterContentInit` cycle. Before refactoring this code, I was running into that very error because at one point I was accessing focusable items too early. But I kept this guard because I found this preemptively necessary.

Reply:
nit: Strange. I removed it locally and see no errors. But you said it was temporary. Was the early access part of a valid scenario? I'd consider the lack of check to be a "fail-fast" guard against unexpected access.
Otherwise we silently cover unexpected behavior.

Reply:
ok I can revert it back.